### DISCUSSION TITLE
List title length limit || Server and client length check work differently

### DISCUSSION THREAD

**Discussion by taep96** [community/community#52449] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

Long emoji names take up a big part of the length limit, while not taking up much space visually (ex. :building_construction: / `:building_construction:`).

---

To work around this issue I've tried 2 things:

1. Inspect element,

2. Pasting the emoji itself, not it's shortcode.

---

I've investigated and found out that on the client the length is limited by setting \<input /> tag's maxlength to 32 - this doesn't count :shortcodes: correctly.

When using inspect element and setting the \<input />'s value directly, then saving, I was able to set a title to something longer than 32.

Using this method, here's some very scientific testing I've done:

1. 31 actual :bug: emojis + 1 required alphanumerical character - success - good.

2. 32 actual :bug: emojis + 1 required alphanumerical character - failure - good.

3. 31 actual :building_construction: emojis + 1 required alphanumerical character - failure - bad.

3. 40 normal characters (`asdfasdfgh` x4) - success - bad.

4. 31 `:bug:` shortcodes - failure - bad.

I suspect that the actual limit on the server is set to a higher number to avoid dealing with the edge cases like this one.

---

In conclusion: 

- I still have no clue how it works
- I wasted a ton of time testing and writing this

To fix:

- Remake the client-side check to count :shortcodes: as a single character
- Or alternatively, remake both checks to use something like [this library](https://crates.io/crates/unicode-width) I've used before to solve a similar problem

---

P.S.

It would be pretty cool if we could drag the lists around like pins on the profile overview page, maybe even place the lists there
